packages/terra-application-layout/src/header/_ApplicationHeader.jsx

```diff
@@ -0,0 +1,253 @@
+            onClick={layoutConfig.toggleMenu}
+          >
+            <IconMenu />
+          </button>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  renderAppName() {
+    const { nameConfig } = this.props;
+
+    if (nameConfig.accessory || nameConfig.title) {
+      return (
+        <ApplicationHeaderName accessory={nameConfig.accessory} title={nameConfig.title} />
+      );
+    }
+
+    return null;
+  }
+
+  renderNavigation(isCompact) {
+    const { applicationLinks } = this.props;
+
+    if (!isCompact) {
+      if (applicationLinks && applicationLinks.length) {
+        return (
+          <ApplicationTabs links={applicationLinks} />
+        );
+      }
+
+      return null;
+    }
+
+    /**
+     * When compact, the navigation region of the header renders the application name component instead.
+     */
+    return this.renderAppName();
+  }
+
+  renderExtensions(isCompact) {
+    const { app, layoutConfig, extensions } = this.props;
+
+    if (!isCompact && extensions) {
+      return React.cloneElement(extensions, { app, layoutConfig });
+    }
+
+    return null;
+  }
+
+  renderUtilities(isCompact) {
+    const { utilityConfig } = this.props;
+
+    if (!isCompact && utilityConfig) {
+      return (
+        <ApplicationHeaderUtility
+          onChange={this.handleUtilityOnChange}
+          onDisclose={this.handleUtilityDiscloseRequest}
+          title={utilityConfig.title}
+          accessory={utilityConfig.accessory}
+          menuItems={utilityConfig.menuItems}
+          initialSelectedKey={utilityConfig.initialSelectedKey}
+          data-application-header-utility
+        />
+      );
+    }
+
+    return null;
+  }
+
+  renderUtilitiesPopup() {
+    const { utilityComponent } = this.state;
+
+    if (utilityComponent) {
+      return (
+        <Popup
+          attachmentBehavior="none"
+          contentAttachment="top center"
+          contentHeight="auto"
+          contentWidth="240"
+          isArrowDisplayed
+          isHeaderDisabled
+          isOpen
+          onRequestClose={this.handleUtilityPopupCloseRequest}
+          targetRef={this.getTargetRef}
+        >
+          {utilityComponent}
+        </Popup>
+      );
+    }
+
+    return null;
+  }
+
+  render() {
+    const {
+      app,
+      applicationLinks,
+      extensions,
+      layoutConfig,
+      nameConfig,
+      utilityConfig,
+      navigationLayoutRoutes,
+      navigationLayoutSize,
+      intl,
+      ...customProps
+    } = this.props;
+
+    const headerClassNames = cx([
+      'application-header',
+      customProps.className,
+    ]);
+
+    const isCompact = Helpers.isSizeCompact(layoutConfig.size);
+
+    return (
+      <div {...customProps} className={headerClassNames} ref={this.setContentNode}>
+        <ApplicationHeaderLayout
+          toggle={this.renderToggle()}
+          logo={!isCompact ? this.renderAppName() : null}
+          navigation={this.renderNavigation(isCompact)}
```

Comment:
It would improve readability if we check `!isCompact` before rending the header, extensions or footer since each of these helpers return null if `isCompact`